cpp/ql/test/query-tests/Likely Bugs/Likely Typos/inconsistentLoopDirection/inconsistentLoopDirection.cpp

```diff
@@ -177,4 +177,22 @@ void FalseNegativeTestCases()
     for (int i = 100; i > 0; i += 2) {}
     // For comparison
     for (int i = 100; i > 0; i ++ ) {}  // BUG
+}
+
+void IntendedOverflow()
+{
+    const unsigned char m = 10;
```

Comment:
When `m` is const, I think expressions like `m - 1` will have a `getValue` result, which means the tests involving `m` are testing constants rather than variables. I'd like to have `m` turned into a parameter instead so the compiler doesn't know its value.

Reply:
The test was intended to be not much more complicated than a constant literal, but I'll add some more cases using a parameter and see what happens (I'm not at all sure they'll be caught).

Reply:
Added - as I suspected, `SimpleRangeAnalysis` isn't up to this.  It only deals with absolute numerical bounds, and it doesn't cope all that well with loops by the look of it.